File: Sample/BasicSample/sample_cpp/Mouse/Mouse_Position.cpp
#include <Altseed.h>

/**
@brief	マウスの座標を取得を取得するサンプル。
*/
void Mouse_Position()
{
	//Altseedを初期化する。
	asd::Engine::Initialize(asd::ToAString("Mouse_Position").c_str(), 640, 480, asd::EngineOption());

	// マウスの状態を表示するテキストを生成する。
	auto font = asd::Engine::GetGraphics()->CreateDynamicFont(asd::ToAString("").c_str(), 25, asd::Color(255, 255, 255, 255), 1, asd::Color(0, 0, 0, 255));

	// マウスカーソルの座標を表示する文字描画オブジェクトを設定して、エンジンに追加する。
	auto stateText = std::make_shared<asd::TextObject2D>();
	stateText->SetPosition(asd::Vector2DF(10, 10));
	stateText->SetFont(font);
	asd::Engine::AddObject2D(stateText);

	// Altseedのウインドウが閉じられていないか確認する。
	while (asd::Engine::DoEvents())
	{
		// マウスカーソルの座標を取得して表示する。
		asd::Vector2DF pos = asd::Engine::GetMouse()->GetPosition();
		stateText->SetText(asd::ToAString(("マウスカーソルの位置 : (" + std::to_string(pos.X) + "," + std::to_string(pos.Y) + ")").c_str()).c_str());
		
		// Altseedを更新する。
		asd::Engine::Update();
	}

	//Altseedの終了処理をする。
	asd::Engine::Terminate();

	return;
}

File: Sample/BasicSample/sample_cs/Mouse/Mouse_Position.cs
/// <summary>
/// マウスの座標を取得を取得するサンプル。
/// </summary>
public class Mouse_Position : ISample
{
    public string Description
    {
        get { return "マウスの座標を取得し、その位置を表示するサンプルです。"; }
    }
	public string Title
	{
		get { return "マウス座標の取得"; }
	}
	public string ClassName
	{
		get { return "Mouse_Position"; }
	}

    public void Run()
    {
        // Altseedを初期化する。
        asd.Engine.Initialize("Mouse_Position", 640, 480, new asd.EngineOption());

        // マウスの状態を表示するテキストを生成する。
        var font = asd.Engine.Graphics.CreateDynamicFont("", 25, new asd.Color(255, 255, 255, 255), 1, new asd.Color(0, 0, 0, 255));

        // マウスカーソルの座標を表示する文字描画オブジェクトを設定して、エンジンに追加する。
        var stateText = new asd.TextObject2D();
        stateText.Position = new asd.Vector2DF(10, 10);
        stateText.Font = font;
        asd.Engine.AddObject2D(stateText);

        // Altseedのウインドウが閉じられていないか確認する。
        while (asd.Engine.DoEvents())
        {
            // マウスカーソルの座標を取得して表示する。
            asd.Vector2DF pos = asd.Engine.Mouse.Position;
            stateText.Text = "マウスカーソルの位置 : (" + pos.X + "," + pos.Y + ")";

            // Altseedを更新する。
            asd.Engine.Update();

            Recorder.CaptureScreen("Mouse_Position", 30, 75, 0.5f, 0.7f);
			Recorder.TakeScreenShot("Mouse_Position", 30);
        }

        //Altseedの終了処理をする。
        asd.Engine.Terminate();
    }
}
